packages/shared/src/components/post/customPrompts/SmartPrompt.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useRef, useState } from 'react';
+import type { ReactElement } from 'react';
+import type { Post } from '../../../graphql/posts';
+import PostSummary from '../../cards/common/PostSummary';
+import { Tab, TabContainer } from '../../tabs/TabContainer';
+import { usePlusSubscription } from '../../../hooks';
+import { PromptButtons } from './PromptButtons';
+import { TargetId } from '../../../lib/log';
+import { PostUpgradeToPlus } from '../../plus/PostUpgradeToPlus';
+import ShowMoreContent from '../../cards/common/ShowMoreContent';
+import { PromptDisplay } from '../../../graphql/prompt';
+
+export const SmartPrompt = ({ post }: { post: Post }): ReactElement => {
+  const { isPlus, showPlusSubscription } = usePlusSubscription();
+  const [activeDisplay, setActiveDisplay] = useState<PromptDisplay>(
+    PromptDisplay.TLDR,
+  );
+  const elementRef = useRef<HTMLDivElement>(null);
+  const width = elementRef?.current?.getBoundingClientRect()?.width || 0;
+
+  const onSetActiveDisplay = (display: PromptDisplay) => {
+    if (!isPlus && display !== PromptDisplay.TLDR) {
+      setActiveDisplay(PromptDisplay.UpgradeToPlus);
+      return;
+    }
+    setActiveDisplay(display);
+  };
+
+  if (!showPlusSubscription) {
+    return <PostSummary className="mb-6" summary={post.summary} />;
+  }
+
+  return (
+    <div
+      className="mb-6 flex flex-col gap-3 text-text-secondary"
+      ref={elementRef}
+    >
+      <PromptButtons
+        activeDisplay={activeDisplay}
+        setActiveDisplay={onSetActiveDisplay}
+        width={width}
+      />
+      <TabContainer<PromptDisplay>
+        controlledActive={activeDisplay}
+        showHeader={false}
+      >
+        <Tab label={PromptDisplay.TLDR}>
+          <ShowMoreContent
+            className="overflow-hidden"
+            content={post.summary}
+            charactersLimit={330}
+            threshold={50}
+          />
+        </Tab>
+        <Tab label={PromptDisplay.SimplifyIt}>simplify</Tab>
```

Comment:
We can load these from the query now.

Reply:
I refactored it a bit. Either it didn't work react wise, or TypeScript weren't happy.
So now there's less available Tabs, as we can actually just re-use them when for streaming the content.